packages/controllers/src/snaps/ReentrancyGuard.ts

```diff
@@ -0,0 +1,26 @@
+export class ReentrancyGuard {
```

Comment:
I don't think this class stands on-its-own as a separate entity outside of `RequestQueue`. Could you just implement the functionality directly into the queue?

Reply:
Going to remove this class, this wasn't preventing the race condition issue that we were trying to guard against. Will think of a different approach.